ovh/resource_cloud_project_database_mongodb_user.go

```diff
@@ -0,0 +1,223 @@
+package ovh
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/ovh/terraform-provider-ovh/ovh/helpers"
+)
+
+func resourceCloudProjectDatabasepMongodbUser() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudProjectDatabaseMongodbUserCreate,
+		Read:   resourceCloudProjectDatabaseMongodbUserRead,
+		Delete: resourceCloudProjectDatabaseMongodbUserDelete,
+		Update: resourceCloudProjectDatabaseMongodbUserUpdate,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudProjectDatabaseMongodbUserImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"service_name": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("OVH_CLOUD_PROJECT_SERVICE", nil),
+			},
+			"cluster_id": {
+				Type:        schema.TypeString,
+				Description: "Id of the database cluster",
+				ForceNew:    true,
+				Required:    true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "Name of the user",
+				ForceNew:    true,
+				Required:    true,
+			},
+			"roles": {
+				Type:        schema.TypeList,
+				Description: "Roles the user belongs to (without authentication database)",
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+
+			//Optional/Computed
```

Comment:
User should not be able to set it's password. This endpoint will be deprecated, we don't want user to get use to use this "feature"